File: Darts/src/burlton/dartzee/code/utils/DartsDebugExtension.java
package burlton.dartzee.code.utils;

import burlton.desktopcore.code.util.ClientEmailer;
import burlton.core.code.util.DebugExtension;
import burlton.desktopcore.code.util.DialogUtil;

public class DartsDebugExtension implements DebugExtension
{
	@Override
	public void exceptionCaught(boolean showError) 
	{
		if (showError)
		{
			DialogUtil.showErrorLater("A serious error has occurred. Logs will now be sent for investigation."
					+ "\nThere is no need to send a bug report.");
		}
	}

	@Override
	public void unableToEmailLogs() 
	{
		DialogUtil.showErrorLater("An error occurred e-mailing logs. Please submit a bug report manually.");
	}

	@Override
	public void sendEmail(String title, String message) throws Exception
	{
		ClientEmailer.sendClientEmail(title, message, false);
	}
}

File: Darts/src/burlton/dartzee/code/utils/DartsDebugExtension.kt
package burlton.dartzee.code.utils

import burlton.dartzee.code.core.util.DebugExtension
import burlton.dartzee.code.core.util.DialogUtil

class DartsDebugExtension : DebugExtension
{
    override fun exceptionCaught(showError: Boolean)
    {
        if (showError)
        {
            DialogUtil.showErrorLater("A serious error has occurred. Logs will now be sent for investigation." + "\nThere is no need to send a bug report.")
        }
    }

    override fun unableToEmailLogs()
    {
        DialogUtil.showErrorLater("An error occurred e-mailing logs. Please submit a bug report manually.")
    }

    override fun sendEmail(title: String, message: String)
    {
        ClientEmailer.sendClientEmail(title, message)
    }
}